src/object/cli_obj.c

```diff
@@ -6506,14 +6507,26 @@ dc_obj_query_key(tse_task_t *api_task)
 
 non_leader:
 		/* Then Try non-leader shards */
+		D_ASSERT(obj_is_ec(obj));
 		start_shard = i * obj_get_grp_size(obj);
 		D_DEBUG(DB_IO, DF_OID" EC needs to try all shards for group %d.\n",
 			DP_OID(obj->cob_md.omd_id), i);
 		for (j = start_shard; j < start_shard + daos_oclass_grp_size(&obj->cob_oca); j++) {
+			if (obj_shard_is_invalid(obj, j, DAOS_OBJ_RPC_QUERY_KEY))
```

Comment:
Does the shard in "reintegrating" is valid or not? obj_shard_is_invalid() seems not exclude such target?

Reply:
it is using oclass_grp_size here, so reintegrating shard is not within this range.